Makefile

```diff
@@ -303,18 +301,19 @@ endif
 # This recipe invokes 'opm' in 'semver' bundle add mode. For more information on add modes, see:
 # https://github.com/operator-framework/community-operators/blob/7f1438c/docs/packaging-operator.md#updating-your-existing-operator
 .PHONY: catalog-build
-catalog-build: opm bundle-push ## Build a catalog image. Bundle image must have been pushed into the registry.
-	$(OPM) index add --container-tool $(CONTAINER_ENGINE) --mode semver --tag $(CATALOG_IMG) --bundles $(BUNDLE_IMGS) $(FROM_INDEX_OPT)
+catalog-build: opm bundle-push ## Generate operator-catalog dockerfile using the operator-bundle image built above; then build catalog image
+	$(OPM) index add --container-tool $(CONTAINER_ENGINE) --mode semver --tag $(CATALOG_IMG) --bundles $(BUNDLE_IMGS) $(FROM_INDEX_OPT) --generate
+	$(CONTAINER_ENGINE) build --platform $(PLATFORM) -f index.Dockerfile -t $(CATALOG_IMG) --label $(LABEL) .
```

Comment:
Just curious, why this change is needed? My local build with docker fails in the catalog-build. Are you able to build it?

Reply:
it's needed so that we can in TWO steps:
* generate the index.Dockefile, then
* build the image and inject the quay expiry label

I've only tested with podman since that's what RH supports. :D Docker is dead, long live podman/buildah/skopeo.

Reply:
I got error below.
export VERSION=0.0.68
make generate manifests operator-build operator-push bundle bundle-build bundle-push catalog-build catalog-push
....
/Users/zhangj/bin/opm index add --container-tool docker --mode semver --tag quay.io/jianrzha/backstage-operator-catalog:0.0.68 --bundles quay.io/jianrzha/backstage-operator-bundle:0.0.68  --generate
WARN[0000] DEPRECATION NOTICE:
Sqlite-based catalogs and their related subcommands are deprecated. Support for
them will be removed in a future release. Please migrate your catalog workflows
to the new file-based catalog format.
INFO[0000] building the index                            bundles="[quay.io/jianrzha/backstage-operator-bundle:0.0.68]"
INFO[0000] running /usr/local/bin/docker pull quay.io/jianrzha/backstage-operator-bundle:0.0.68  bundles="[quay.io/jianrzha/backstage-operator-bundle:0.0.68]"
INFO[0001] running docker create                         bundles="[quay.io/jianrzha/backstage-operator-bundle:0.0.68]"
INFO[0002] running docker cp                             bundles="[quay.io/jianrzha/backstage-operator-bundle:0.0.68]"
ERRO[0002] Error response from daemon: No such container: WARNING  bundles="[quay.io/jianrzha/backstage-operator-bundle:0.0.68]"
ERRO[0002] permissive mode disabled                      bundles="[quay.io/jianrzha/backstage-operator-bundle:0.0.68]" error="error copying container directory Error response fr

Reply:
with help from https://www.digitalocean.com/community/questions/how-to-fix-docker-got-permission-denied-while-trying-to-connect-to-the-docker-daemon-socket I can now run...

```
make release-build
... 
docker build --platform linux/amd64 -f docker/Dockerfile -t quay.io/janus/operator:0.0.1 --label quay.expires-after=14d .
...
Step 1/19 : FROM registry.access.redhat.com/ubi9/go-toolset:1.20.10-6 as builder
...
```